### PR TITLE
Fixes bot runtimes, and makes secbots no longer see through walls and lockers

### PR DESCRIPTION
**Pull request by Superlagg**
## About the PR 

Secbots now check if the target is in view before engaging them. They also give up and refuse to attack people hiding within things, so you can duck inside a locker now to escape Beepsky's wrath.

Similar thing for Medbots. They can still see through walls, but they'll refuse to medicate someone if they're within something, like a locker or clone tube.

Also added a few more messages when secbots are forced to give up.

Also changed up the bot mover to work directly from being initialized, and hopefully not runtime quite so much anymore.

Also fixes bots runtiming while inside things or saying things weirdly.

## Why's this needed? 

Shit busted, now less busted. #3842 

## Changelog

```changelog
(u)Superlagg
(*)Securitrons will no longer attack you if you're tucked away inside something.
```

### CODE REVIEW

**Comment by Avatharian** [comment]
As someone who recently spent 15 minutes getting the absolute _shit_ kicked out of her by double-emagged Beepsky while inside a locker, this makes me happy.

File: code/modules/robotics/bot/secbot.dm [hunk 1]
```diff
@@ -803,16 +809,11 @@
 	// look for a criminal in range of the bot
 	proc/look_for_perp()
 		src.anchored = 0
-		for_by_tcl(C, /mob/living/carbon) //Let's find us a criminal
-			if(!IN_RANGE(src, C, 7)) // We've made a plea bargain with opaque objects to turn in criminals hiding behind them
-				continue
+		for(var/mob/living/carbon/C in view(7, get_turf(src))) //Let's find us a criminal
```
Comment (ZeWaka) [inline code/modules/robotics/bot/secbot.dm:812]: GET_NEARBY ?

File: code/modules/robotics/bot/bot_parent.dm [hunk 2]
```diff
@@ -263,70 +264,85 @@
...
 		return
 
 	disposing()
-		if(master.bot_mover == src)
-			master.bot_mover = null
-		master.moving = 0
+		if(istype(master))
+			if(master?.bot_mover == src)
+				master.bot_mover = null
+			master?.moving = 0
```
Comment (ZeWaka) [inline code/modules/robotics/bot/bot_parent.dm:307]:
```suggestion
			if(master.bot_mover == src)
				master.bot_mover = null
			master.moving = FALSE
```
a typed master can't be niull

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/robotics/bot/secbot.dm (hunk 1 above)
Comment (Superlagg) [reply code/modules/robotics/bot/secbot.dm:812]: That'd make them able to see through walls, which I'm trying to fix.

**Pull request merged**